src/Poseidon/Janno.hs

```diff
@@ -1062,33 +1111,50 @@ allEqual :: Eq a => [a] -> Bool
 allEqual [] = True
 allEqual x  = length (nub x) == 1
 
-checkC14ColsConsistent :: JannoRow -> Bool
+checkC14ColsConsistent :: JannoRow -> JannoRowLog JannoRow
 checkC14ColsConsistent x =
-    let lLabnr          = getCellLength $ jDateC14Labnr x
-        lUncalBP        = getCellLength $ jDateC14UncalBP x
-        lUncalBPErr     = getCellLength $ jDateC14UncalBPErr x
-        shouldBeTypeC14 = lUncalBP > 0
-        isTypeC14       = jDateType x == Just C14
-    in
-        (lLabnr == 0 || lUncalBP == 0 || lLabnr == lUncalBP)
-        && (lLabnr == 0 || lUncalBPErr == 0 || lLabnr == lUncalBPErr)
-        && lUncalBP == lUncalBPErr
-        && (not shouldBeTypeC14 || isTypeC14)
-
-checkContamColsConsistent :: JannoRow -> Bool
+    let isTypeC14        = jDateType x == Just C14
+        lLabnr           = getCellLength $ jDateC14Labnr x
+        lUncalBP         = getCellLength $ jDateC14UncalBP x
+        lUncalBPErr      = getCellLength $ jDateC14UncalBPErr x
+        anyMainColFilled = lUncalBP > 0 || lUncalBPErr > 0
+        anyMainColEmpty  = lUncalBP == 0 || lUncalBPErr == 0
+        allSameLength    = allEqual [lLabnr, lUncalBP, lUncalBPErr] ||
+                          (lLabnr == 0 && lUncalBP == lUncalBPErr)
+    in case (isTypeC14, anyMainColFilled, anyMainColEmpty, allSameLength) of
+        (False, False, _, _ )   -> return x
+        (False, True, _, _ )    -> E.throwError "Date_Type is not \"C14\", but either \
+                                         \Date_C14_Uncal_BP or Date_C14_Uncal_BP_Err are not empty"
+        (True, _, False, False) -> E.throwError "Date_C14_Labnr, Date_C14_Uncal_BP and Date_C14_Uncal_BP_Err \
+                                         \do not have the same lengths. Date_C14_Labnr can be empty"
```

Comment:
I think the addition "Date_C14_Labnr can be empty" is not helping here. I would remove it. The error concerns the non-equal length, and the error should focus only on that. The positive message coming after muddles the main message, in my view. Otherwise you would also have to write it in the last case below, I think?

Reply:
Removed.